### PR TITLE
Chore/tweaks for v1

### PR DESCRIPTION
**Pull request by aesopwolf**
I'm going to push changes to this PR over the next few days / week in preparation for releasing v1.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "airbnb"
```
Comment (track0x1) [inline .eslintrc]: Yes!

File: README.md [hunk 2]
```diff
@@ -1,124 +1,132 @@
...
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.disableButton = this.disableButton.bind(this);
+    this.enableButton = this.enableButton.bind(this);
+    this.submit = this.submit.bind(this);
+    this.state = { canSubmit: false };
+  }
+
+  disableButton = () => {
```
Comment (track0x1) [inline README.md]: Use ES7 class property + arrow function (as is here), or bind in the constructor (L81)... I've preferred arrow functions so you can easily see what functions are being bound when they are declared as opposed to having to look in the constructor. 🙂

File: README.md [hunk 3]
```diff
@@ -1,124 +1,132 @@
...
+    this.setState({ canSubmit: false });
+  }
+
+  enableButton = () => {
```
Comment (track0x1) [inline README.md]: Same comment as above applies here.

File: README.md [hunk 4]
```diff
@@ -1,124 +1,132 @@
...
+import Formsy from 'formsy-react';
+import React from 'react';
+import MyInput from './MyInput';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.disableButton = this.disableButton.bind(this);
+    this.enableButton = this.enableButton.bind(this);
+    this.submit = this.submit.bind(this);
```
Comment (track0x1) [inline README.md]: Do you need to bind `submit`? It doesn't appear to be using `this` in it.

File: package.json [hunk 5]
```diff
@@ -1,48 +1,61 @@
...
+    "react-addons-pure-render-mixin": "^15.6.0",
+    "react-addons-test-utils": "^15.6.0",
+    "react-dom": "^15.6.1",
+    "sinon": "^3.2.0",
+    "webpack": "^3.5.4",
+    "webpack-dev-server": "^2.7.1"
+  },
+  "peerDependencies": {
+    "react": "^15.6.1",
+    "react-dom": "^15.6.1"
```
Comment (track0x1) [inline package.json]: This is listed on L52 as a dev dependency as well.

Comment (aesopwolf) [inline package.json]: It needs to be kept as a dev dependency for running both the tests and examples, no?

**Comment by aesopwolf** [comment]
@track0x1 I'm running into a problem with the reset-values example. Would you mind taking a look at it? Just run `yarn examples`, open up `http://localhost:8080/reset-values/index.html` and click the "Reset" button -- I can't figure out why the checkbox disappears. I've been trying to debug this for 2 days. 

I'm thinking about using "defaultChecked" in `Checkbox.js` but will need to wait for react v15.6.2 since there's [a bug at the moment](https://github.com/facebook/react/issues/9988)

**Comment by track0x1** [comment]
@aesopwolf Looks like its caused by the behavior of the browsers `<form>` API
>type `reset`: The button resets all the controls to their initial values.

The problem is that in a field like a checkbox, you have to actually change the value of the `checked` attribute to check/uncheck it, which removes our initial state that would otherwise be used by `reset()` provided via form API.

Easiest solution is to avoid using `reset` button type and switch it to something like `type="button"`.

File: package.json (hunk 5 above)
Comment (track0x1) [inline package.json]: You're right! I've made a habit of using dependencies + devDependencies, ditching peer deps altogether 🙃, but this is the better route.

**Comment by aesopwolf** [comment]
Wow... I have no words. Thank you so much. I was going crazy.

**Comment by aesopwolf** [comment]
It's been a while since I've touched this. I'm just going to merge it in to the beta branch as is and open up more PR's as need be.

**Pull request merged**